backends-clickhouse/src/main/scala/org/apache/gluten/extension/RewriteDateTimestampComparisonRule.scala

```diff
@@ -37,7 +37,7 @@ import org.apache.spark.unsafe.types.UTF8String
 // This rule try to make the filter condition into integer comparison, which is more efficient.
 // The above example will be rewritten into
 //    select * from table where to_unixtime('2023-11-02', 'yyyy-MM-dd') >= unix_timestamp
-class RewriteDateTimestampComparisonRule(session: SparkSession, conf: SQLConf)
```

Comment:
Thanks, this kind of simplification is needed.

Reply:
Though, do we really need `session` as a parameter to create `GlutenConfig`? What's the issue of using `GlutenConfig.getConf`?

Reply:
If there is an active `SparkSession`, the proper `SQLConf` associated with the thread's active session is used. So we should use `spark.sessionState.conf` first.
`Glutenconfig.getConf` actually proxies `SQLConf.get`, which will prioritize obtaining SQLConf from the active session, otherwise the default SQLConf will be used.

Reply:
> Glutenconfig.getConf actually proxies SQLConf.get

Spark has `SQLConfHelper` that a lot of components (including `Rule`) rely on, using `SQLConf.get` to access session config. Is there any special consideration to avoid using it in Gluten?

Reply:
No. This PR doesn't avoid using `SQLConf.get` in Gluten.
In fact, `SQLConf.get` and `spark.sessionState.conf` are functionally equivalent. But `spark.sessionState.conf` is faster than `SQLConf.get` because the latter requires visit thread local. So we can see the usage scenarios:

1. The code rely on `SQLConfHelper` in Spark we can't find `SparkSession` in context.
2. All the places in Spark rely on `spark.sessionState.conf` if exists `SparkSession`.